File: 401.py
class Solution:
    def readBinaryWatch(self, turnedOn: int) -> List[str]:
        # time: O(1), space: O(1)
        if turnedOn < 0 or 8 < turnedOn:
            return []
        res = []
        for h in range(12):
            for m in range(60):
                if bin(h).count("1") + bin(m).count("1") == turnedOn:
                    res.append("{}:{:02d}".format(h, m))
        return res

File: 401.java
class Solution {
    public List<String> readBinaryWatch(int turnedOn) {
        // time: O(1), space: O(1)
        List<String> times = new ArrayList<>();
        for (int h = 0; h < 12; h++) {
            for (int m = 0; m < 60; m++) {
                if (Integer.bitCount(h * 64 + m) == turnedOn) {
                    times.add(String.format("%d:%02d", h, m));
                }
            }
        }
        return times;
    }
}